Add verify_end_to_end_purchase.py

jasnoludek opened the pull request.

jasnoludek committed:
Add verify_end_to_end_purchase.py

techcoachralph commented on Tests/verify_end_to_end_purchase.py:
```diff
@@ -0,0 +1,175 @@
+from selenium.webdriver.chrome.service import Service as ChromeService
+from webdriver_manager.chrome import ChromeDriverManager
+
+from Pages.home_page import HomePage
```
I would put Pages in lower case as it is a package. `pages.home_page`

jasnoludek replied:
Pages package renamed to pages per naming conventions.

techcoachralph commented on Tests/verify_end_to_end_purchase.py:
```diff
@@ -0,0 +1,175 @@
+        login.enter_password('MyPassword1234$')
+        login.click_login()
+
+        time.sleep(2)
```
I would recommend using a WebDriverWait function here. It's not good to use time.sleep when automating

jasnoludek replied:
WebDriverWait replaces time.sleep where necessary.

techcoachralph commented on Tests/verify_end_to_end_purchase.py:
```diff
@@ -0,0 +1,175 @@
+        time.sleep(2)
+
+        # Navigate back to home page
+        myaccount = MyAccountPage(browser)
```
Recommendation to group instantiating page objects at the top of the test so you don't have to do them through the test. It is cleaner and easier to read

jasnoludek replied:
Page objects now instantiated at top of the test.

techcoachralph reviewed (changes requested):
This was a GREAT job! The only serious feedback would be to change the time.sleep to a WebDriver wait function.

jasnoludek committed:
Refractor end to end purchase test

This commit makes changes to the test case as suggested in code review.

The package name homepage was converted to lower case as per naming
conventions.

Page classes are now defined at the top of the test script rather than
in multiple sections repeatedly. This makes for better organization
and cleaner code.

WebDriverWait is utilized rather than time.sleep.

Changes as above will be applied to other test cases in this project
after approval.

techcoachralph commented on Locators/locators.py:
```diff
@@ -97,6 +97,6 @@ class Locators():
 
     # Order received confirmation
-    order_received_confirmation_css = '//*[@id="post-8"]/div/div/div/p'
+    order_received_confirmation_xpath = '//*[@id="post-8"]/div/div/div/p'
```
this is not the safest xpath. It looks like it can break easily

jasnoludek committed:
Add test case for searching hoodies on the home page

This commit introduces a new test case to verify the functionality of searching for
hoodies on the home page of the SuperSQA Demo Store website. The test case uses
Selenium for automated testing.

Test Case Details:
- Test Class: TestSearchForHoodies
- Test Method: test_verify_search
- Actions:
  - Open the Chrome browser using ChromeDriverManager
  - Navigate to the SuperSQA Demo Store website
  - Utilize the HomePage class to search for the term "hoodie"
  - Locate and count the search results displayed in a specific section
  - Assert that the count of search results matches the expected count
  - Perform cleanup by closing the browser

jasnoludek committed:
Add test case for searching hoodies on the home page
This commit introduces a new test case to verify the functionality of searching for
hoodies on the home page of the SuperSQA Demo Store website. The test case uses
Selenium for automated testing.

Test Case Details:
- Test Class: TestSearchForHoodies
- Test Method: test_verify_search
- Actions:
  - Open the Chrome browser using ChromeDriverManager
  - Navigate to the SuperSQA Demo Store website
  - Utilize the HomePage class to search for the term "hoodie"
  - Locate and count the search results displayed in a specific section
  - Assert that the count of search results matches the expected count
  - Perform cleanup by closing the browser
